pkg/sshutil/sshutil.go

```diff
@@ -261,3 +263,26 @@ func detectOpenSSHVersion() semver.Version {
 	}
 	return v
 }
+
+// detectValidPublicKey returns whether content represent a public key.
+// OpenSSH public key format have the structure of '<algorithm> <key> <comment>'.
+// By checking 'algorithm' with signature format identifier in 'key' part,
+// this function may report false positive but provide better compatibility.
+func detectValidPublicKey(content string) bool {
+	// content have been space-trimmed
```

Comment:
I would prefer to keep the check for a newline character in here. We added it because having a newline breaks the generated cloud-init YAML and doesn't allow connecting to the instance via SSH at all.

I guess this would now be caught by the signature check anyways, but keeping the check costs little, even if it may be redundant.

Reply:
Looks good to me, branch added